Distiller-server/src/main/java/com/mapr/distiller/server/metricactions/SystemCpuMetricAction.java

```diff
@@ -0,0 +1,162 @@
+import com.mapr.distiller.server.queues.RecordQueueManager;
+import com.mapr.distiller.server.recordtypes.Record;
+import com.mapr.distiller.server.recordtypes.SystemCpuRecord;
+import com.mapr.distiller.server.utils.Constants;
+import com.mapr.distiller.server.utils.MetricConfig;
+
+public class SystemCpuMetricAction extends MetricAction implements
+		MetricsSelectable {
+	Object object = new Object();
+
+	private RecordQueue inputQueue;
+	private RecordQueue outputQueue;
+	private String recordType;
+
+	private SystemCpuRecordProcessor recordProcessor;
+
+	private String aggregationType;
+	private Map<String, String> aggregationMap;
+
+	private boolean shouldPersist;
+
+	public SystemCpuMetricAction(String id, String recordType,
+			String aggregationType, Map<String, String> aggregationMap,
+			RecordQueue inputQueue, RecordQueue outputQueue) {
+		super(id);
+		this.recordType = recordType;
+		this.aggregationType = aggregationType;
+		this.aggregationMap = aggregationMap;
+
+		this.inputQueue = inputQueue;
+		this.outputQueue = outputQueue;
+
+		setGathericMetric(true);
+	}
+
+	// We also need one more argument which has a RecordQueue map to their names
+	public static SystemCpuMetricAction getInstance(MetricConfig metricConfig,
+			RecordQueueManager queueManager) {
+		return new SystemCpuMetricAction(metricConfig.getId(),
+				metricConfig.getRecordType(),
+				metricConfig.getAggregationType(),
+				metricConfig.getAggregationMap(),
+				queueManager.getQueue(metricConfig.getInputQueue()),
+				queueManager.getQueue(metricConfig.getOutputQueue()));
+	}
+
+	@Override
+	public void run() {
+		System.out.println("****" + this.recordType + "****");
+		while (!Thread.interrupted()) {
+			if (isGathericMetric()) {
+				selectSequentialRecords();
+				/*try {
+					Thread.sleep(1000);
+				} catch (InterruptedException e) {
+					System.out.println("Thread got interrupted - Going down");
+					break;
+				}*/
+			}
+
+			else {
+				try {
+					synchronized (object) {
+						object.wait();
+					}
+				} catch (InterruptedException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+			}
+		}
+
+		System.out.println("Exiting Metric action due to kill action");
+	}
+
+	@Override
+	public void suspend() throws InterruptedException {
+		System.out.println("Stopping metric with id = " + id);
+		if (isGathericMetric) {
+			isGathericMetric = false;
+		}
+
+		else {
+			System.out.println("Already suspended metric " + id);
+		}
+	}
+
+	@Override
+	public void resume() {
+		synchronized (object) {
+			if (!isGathericMetric) {
+				System.out.println("Resuming metric with id = " + id);
+				isGathericMetric = true;
+				object.notifyAll();
+			}
+
+			else {
+				System.out.println("Already running Metric " + id);
+			}
+		}
+	}
+
+	@Override
+	public void kill() {
+		System.out.println("Kill metric with id = " + id);
+		isGathericMetric = false;
+	}
+
+	@Override
+	public void selectSequentialRecords() {
```

Comment:
Select sequential records is going to do the same action regardless of the actual record type.  I see this section of code in a class for SystemCpu, a class for SystemMemory.  This needs to be generic for Record extending classes instead of bound to a specific record type.  There will need to be multiple additional selection algorithms (as mentioned in my email earlier today). 

Is this code as it is now just temporary?  Are you planning to make the mentioned change?